src/Auth.ts

```diff
@@ -320,13 +330,24 @@ export class Auth {
 
         // convert a PKCS#8 ASN.1 PrivateKeyInfo to PEM
         cert = pki.privateKeyInfoToPem(privateKeyInfo);
+
+        if (thumbprint === undefined) {
+          const certBags: {
+            [key: string]: pkcs12.Bag[] | undefined;
+            localKeyId?: pkcs12.Bag[];
+            friendlyName?: pkcs12.Bag[];
+          } = p12Parsed.getBags({bagType: pki.oids.certBag});
+          const certBag: pkcs12.Bag = (certBags[pki.oids.certBag] as pkcs12.Bag[])[0];
+          
+          thumbprint = this.calculateThumbprint(certBag.cert as pki.Certificate);
+        }
```

Comment:
After calculating the thumbprint ourselves, we should persist it on the service, so that we don't need to calculate it each time and can rather reuse what we calculated initially

Reply:
Ok. Made some minor tweaks to eliminate the local variable, as it is useless now.